docs/release.md

```diff
@@ -0,0 +1,48 @@
+# Release process
+
+This project follows [semantic versioning](https://semver.org/). The following
+documentation will refer to `X.Y.Z` as _major_, _minor_ and _patch_ version.
+
+## Development between releases
+
+- Every substantial pull request should add an entry to the `[unreleased]`
+  section of the corresponding crate `CHANGELOG.md` file. See
+  [#1698](https://github.com/libp2p/rust-libp2p/pull/1698/files) as an example.
+  
+  In case there is no `[unreleased]` section yet, create one with an increased
+  major, minor or patch version depending on your change. In addition update the
+  corresponding entry of the crate in the root level `Cargo.toml`.
+
+
+## Releasing one or more crates
+
+1. Set the release date for each crate to be released in the respective
+   `CHANGELOG.md` by replacing `# X.Y.Z [unreleased]` with
+   `# X.Y.Z [yyyy-mm-dd]` and create a pull request against the rust-libp2p
+   `master` branch.
+
+2. Once merged, create and push a tag for each updated crate.
+
+    ```
+    $ tag="<CRATE-NAME>-X.Y.Z"
+    $ git tag -s "${tag}" -m "${tag}"
```

Comment:
I have never created a GPG signed tag so far, but went for annotated tags. I also create them all locally and eventually push them altogether  with `git push origin master --tags`.

Reply:
Added a comment above. Let me know if that looks good to you @twittner.